Recruit-Training/Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -1,38 +1,68 @@
 #include "BatteryStateOfChargeService.h"
+#include "BatteryData.h"
+
+#define HOURS_TO_MS (3600.0 * 1000.0)
 
 namespace
 {
     const double BATTERY_AMP_HOUR_CAPACITY = 123.0;
 }
 
+static double prevCurrent, current, voltage, totalAmpHours;
+static QTime prevTime, currentTime;
+
+//static QTime timeUntilCharged, timeUntilDepleted;
+
+//Constructors
 BatteryStateOfChargeService::BatteryStateOfChargeService(double initialStateOfChargePercent)
 : initialStateOfChargePercent_(initialStateOfChargePercent)
 {
+    initialAmpHours_ = BATTERY_AMP_HOUR_CAPACITY * (initialStateOfChargePercent/100);
+    prevTime = QTime(0,0);
+    totalAmpHours = initialAmpHours_;
```

Comment:
totalAmpHours should be renamed to totalAmpHoursUsed, if that is the intention. If it is the intention, then this assignment is actually incorrect (totalAmpHoursUsed should be the Capacity - the initialAmpHours)